### PR TITLE
Wip 832 hammer

### PR DESCRIPTION
**Pull request by smithfarm**
This backports #556, #832 and #858 to hammer

Fixes: http://tracker.ceph.com/issues/16769
Fixes: http://tracker.ceph.com/issues/16641

### CODE REVIEW

**Comment by dachary** [comment]
<pre>
Reviewed-by: Loic Dachary &lt;ldachary@redhat.com>
</pre>

after a run of a relavant rados job confirms is works as expected

**Comment by smithfarm** [comment]
@dachary This run hopefully qualifies: http://pulpito.ceph.com/smithfarm-2016-07-21_14:08:16-rados-hammer-backports---basic-smithi/

**Comment by smithfarm** [comment]
Ran into trouble. The job I want to run is `rados/singleton/{all/divergent_priors.yaml fs/btrfs.yaml msgr-failures/few.yaml}`

The command I'm trying is `./virtualenv/bin/teuthology-suite --filter="rados/singleton/{all/divergent_priors.yaml fs/btrfs.yaml msgr-failures/few.yaml}" --priority 101 --suite rados --suite-branch wip-832-hammer --email ncutler@suse.cz --ceph hammer-backports --machine-type smithi`

The result is: 3237/3237 jobs were filtered out.

Any idea what I'm doing wrong?

File: suites/rados/singleton/all/ec-lost-unfound-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- - mon.a
+  - mon.b
+  - mon.c
+  - osd.0
+  - osd.1
+  - osd.2
+- - osd.3
+tasks:
+- install:
+    branch: infernalis
```
Comment (smithfarm) [inline suites/rados/singleton/all/ec-lost-unfound-upgrade.yaml]: @dachary This commit introduces tests that install infernalis and then upgrade, so I guess the commit should be dropped from this PR. Agree?

**Comment by smithfarm** [comment]
These tests appear to be broken by 139c100 - @dachary shall we drop it?

* http://158.69.66.4:8081/ubuntu-2016-07-23_21:53:47-rados:singleton-hammer-backports---basic-openstack/
* http://158.69.66.4:8081/ubuntu-2016-07-23_21:52:44-rados:singleton-hammer-backports---basic-openstack/

**Comment by smithfarm** [comment]
Changelog:

* dropped backport of ef75c89 from the patchset
* dropped backport of b2a08ef from the patchset

**Comment by smithfarm** [comment]
@dachary Here's a rados run with the latest changes:

http://158.69.66.4:8081/ubuntu-2016-07-24_20:00:22-rados-hammer-backports---basic-openstack/

**Pull request merged**